src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/CartonServiceTest.java

```diff
@@ -1,144 +1,270 @@
+
+        CartonDTO cartonDTO1 = new CartonDTO();
+        cartonDTO1.setId(1L);
+        cartonDTO1.setName("Carton 1");
+
+        CartonDTO cartonDTO2 = new CartonDTO();
+        cartonDTO2.setId(2L);
+        cartonDTO2.setName("Carton 2");
+
+        when(modelMapper.map(carton1, CartonDTO.class)).thenReturn(cartonDTO1);
+        when(modelMapper.map(carton2, CartonDTO.class)).thenReturn(cartonDTO2);
+
+        List<CartonDTO> result = cartonService.getAllCartons();
+
+        assertEquals(mockCartons.size(), result.size());
+        assertEquals(mockCartons.get(0).getName(), result.get(0).getName());
+        assertEquals(mockCartons.get(1).getName(), result.get(1).getName());
+
+        verify(cartonRepository, times(1)).findByDeletedFalse();
+
+        verify(modelMapper, times(mockCartons.size())).map(any(Carton.class), eq(CartonDTO.class));
     }
+    @Test
+    public void testGetCartonById_ExistingId() throws ChangeSetPersister.NotFoundException {
+        Carton carton = new Carton();
+        carton.setId(1L);
+        carton.setName("Carton 1");
+
+        when(cartonRepository.findByIdAndDeletedFalse(1L)).thenReturn(Optional.of(carton));
+
+        CartonDTO cartonDTO = new CartonDTO();
+        cartonDTO.setId(1L);
+        cartonDTO.setName("Carton 1");
+
+        when(modelMapper.map(carton, CartonDTO.class)).thenReturn(cartonDTO);
 
+        CartonDTO result = cartonService.getCartonById(1L);
+
+        assertEquals(cartonDTO.getId(), result.getId());
+        assertEquals(cartonDTO.getName(), result.getName());
+
+        verify(cartonRepository, times(1)).findByIdAndDeletedFalse(1L);
+        verify(modelMapper, times(1)).map(carton, CartonDTO.class);
+    }
 
     @Test
-    void testGetCartonById_ExistingCarton() {
-        Carton cartonEntity = new Carton();
-        cartonEntity.setId(999L);
-        cartonEntity.setName("Carton 1");
-        cartonEntity.setAvailableQuantity(10);
-        when(cartonRepository.findById(any(Long.class))).thenReturn(Optional.of(cartonEntity));
-        when(cartonMapper.toDto(any(Carton.class))).thenReturn(new CartonDTO());
-        CartonDTO resultDto = cartonService.getCartonById(999L);
-        assertNotNull(resultDto);
+    public void testGetCartonById_NonExistingId() {
+        when(cartonRepository.findByIdAndDeletedFalse(1L)).thenReturn(Optional.empty());
+        assertThrows(ChangeSetPersister.NotFoundException.class, () -> cartonService.getCartonById(1L));
 
+        verify(cartonRepository, times(1)).findByIdAndDeletedFalse(1L);
+        verifyNoInteractions(modelMapper);
     }
 
     @Test
-    void testGetCartonById_NonExistingCarton() {
-        when(cartonRepository.findById(any(Long.class))).thenReturn(Optional.empty());
-        assertThrows(ApiRequestException.class, () -> {
-            cartonService.getCartonById(1L);
-        });
+    public void testCreateCarton_InvalidCartonDTO_NameMissing() {
```

Comment:
Do we have a test method that checks for a successful creation of Carton?

Reply:
Yes, we have that test but in other branch which is also in pull-request